### PR TITLE
Add `rawtx` in getTransactionUrl for DeFiScanProvider

### LINKED ISSUE #517: Inclusive of `?rawtx=hex` in getUrl for `DeFiScanProvider`
#### What would you like to be added:

Currently in DeFiScanProvider, `https://defiscan.live/transactions/:txid` is the canonical URL for showing transaction of DeFi Scan. Would like a new query string `?rawtx=hex` to be added where hex represents the transaction encoded in hex.

#### Why is this needed:

This allows DeFi Scan to display rich transaction information before they get confirmed.

/triage accepted
/priority urgent-now
/area contexts transactions

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- To allow DeFi Scan to display rich transaction information before they get confirmed.

#### Which issue(s) does this PR fixes?:

Fixes #517

#### Additional comments?:
Tested on iPhone 12 and Pixel 3a emulator

### CODE REVIEW

File: app/contexts/DeFiScanContext.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { EnvironmentNetwork } from '../environment'
 import { useNetworkContext } from './NetworkContext'
 
 interface DeFiScan {
-  getTransactionUrl: (txid: string) => string
+  getTransactionUrl: (txid: string, hexTxId?: string) => string
```
Comment (fuxingloh) [inline app/contexts/DeFiScanContext.tsx:6]:
```suggestion
  getTransactionUrl: (txid: string, rawtx?: string) => string
```

File: app/contexts/DeFiScanContext.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function DeFiScanProvider (props: React.PropsWithChildren<any>): JSX.Elem
   )
 }
 
-function getTxURLByNetwork (network: EnvironmentNetwork, txid: string): string {
+function getTxURLByNetwork (network: EnvironmentNetwork, txid: string, hexTxId?: string): string {
```
Comment (fuxingloh) [inline app/contexts/DeFiScanContext.tsx:33]:
```suggestion
function getTxURLByNetwork (network: EnvironmentNetwork, txid: string, rawtx?: string): string {
```

File: app/contexts/DeFiScanContext.tsx [hunk 3]
```diff
@@ -39,14 +39,18 @@ function getTxURLByNetwork (network: EnvironmentNetwork, txid: string): string {
...
 
     case EnvironmentNetwork.LocalPlayground:
     case EnvironmentNetwork.RemotePlayground:
-      baseUrl = `${baseUrl}?network=${EnvironmentNetwork.RemotePlayground}`
+      baseUrl += `?network=${EnvironmentNetwork.RemotePlayground}`
       break
   }
 
+  if (typeof hexTxId === 'string' && hexTxId.length !== 0) {
+    baseUrl += `&rawtx=${hexTxId}`
```
Comment (fuxingloh) [inline app/contexts/DeFiScanContext.tsx:52]: How about for main net handling where ? wasn't prefixed?

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: app/contexts/DeFiScanContext.tsx (hunk 3 above)
Comment (kyleleow) [reply app/contexts/DeFiScanContext.tsx:52]: Does mainnet uses a different rawtx value?What are the special

Comment (fuxingloh) [reply app/contexts/DeFiScanContext.tsx]: MainNet does comes with the ?network=MainNet making your query string invalid

Comment (kyleleow) [reply app/contexts/DeFiScanContext.tsx]: Got what you mean now, sorry will update again

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**